src/PKSim.Core/Services/ParameterSetUpdater.cs

```diff
@@ -150,16 +153,27 @@ public ICommand UpdateValuesByName(IEnumerable<IParameter> sourceParameters, IEn
          return updateCommands;
       }
 
-      public ICommand UpdateValue(IParameter sourceParameter, IParameter targetParameter, bool updateParameterOriginId=true)
+      public ICommand UpdateValue(IParameter sourceParameter, IParameter targetParameter, bool updateParameterOriginId = true)
       {
-         if(updateParameterOriginId)
+         if (updateParameterOriginId)
             _parameterIdUpdater.UpdateParameterId(sourceParameter, targetParameter);
 
-         var command = _parameterUpdater.UpdateValue(sourceParameter, targetParameter);
+         return withUpdatedValueOrigin(_parameterUpdater.UpdateValue(sourceParameter, targetParameter), sourceParameter, targetParameter);
+      }
+
+      private ICommand withUpdatedValueOrigin(ICommand command, IParameter sourceParameter, IParameter targetParameter)
+      {
+         if (sourceParameter.ValueOrigin.IsIdenticalTo(targetParameter.ValueOrigin))
+            return command;
+
+         var updateValueOriginCommand = _parameterTask.SetParameterValueOrigin(targetParameter, sourceParameter.ValueOrigin);
+         if (command == null)
```

Comment:
command may be null if ONLY value origin has changed. In that case return the valueorigincommand